modules/blockbuilder/tenant_store.go

```diff
@@ -88,15 +88,35 @@ func (s *tenantStore) AppendTrace(traceID []byte, tr []byte, ts time.Time) error
 	return nil
 }
 
-func (s *tenantStore) Flush(ctx context.Context, store tempodb.Writer) error {
+func (s *tenantStore) Flush(ctx context.Context, r tempodb.Reader, w tempodb.Writer, c tempodb.Compactor) error {
 	if s.liveTraces.Len() == 0 {
 		// This can happen if the tenant instance was created but
 		// no live traces were successfully pushed. i.e. all exceeded max trace size.
 		return nil
 	}
 
+	blockID, existingBlocksToBeCompacted, err := s.determineBlockIDs(ctx, r)
+	if err != nil {
+		return err
+	}
+
+	// TODO - Check if the existing block can be reused and exit early
+
+	if len(existingBlocksToBeCompacted) > 0 {
+		for _, blockID := range existingBlocksToBeCompacted {
+			level.Info(s.logger).Log(
```

Comment:
one last thing. should this be a warning instead? I wonder if we should add a metric to alert us when this happens. This should be a really edge case, otherwise the bug is on our side

Reply:
Wouldn't go so far as to call it a bug, since any restart mid-way through a cycle can cause it.  My guess is that even with changes #4870 and #4874, it will be less frequent, but still a regular occurrence. But good with changing to warning.